tensorflow_probability/python/distributions/empirical_test.py

```diff
@@ -0,0 +1,694 @@
+      self.assertAllEqual(samples_shape[:-2],
+                          self.evaluate(dist.batch_shape_tensor()))
+      self.assertAllEqual(samples_shape[-1:],
+                          self.evaluate(dist.event_shape_tensor()))
+
+  def testCdfNoBatch(self):
+    sample = [[0, 0], [0, 1], [0, 1], [1, 2]]
+    events = [
+        [0, 1],
+        [[0, 1], [1, 2]],
+        [[[0, 0], [0, 1]],
+         [[1, 1], [1, 2]]]
+    ]
+    expected_cdfs = [
+        [0.75, 0.75],
+        [[0.75, 0.75],
+         [1, 1]],
+        [[[0.75, 0.25], [0.75, 0.75]],
+         [[1., 0.75], [1., 1.]]]
+    ]
+
+    for event, expected_cdf in zip(events, expected_cdfs):
+      input_ = tf.convert_to_tensor(sample, dtype=np.float32)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=input_.shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph, event_ndims=1)
+      self.assertAllClose(self.evaluate(dist.cdf(event)),
+                          expected_cdf)
+      self.assertAllClose(self.evaluate(dist.log_cdf(event)),
+                          np.log(expected_cdf))
+
+  def testCdfWithBatch(self):
+    sample = [[[0, 0], [0, 1], [0, 1], [1, 1]],
+              [[0, 1], [2, 2], [2, 2], [2, 2]]]
+    events = [
+        [0, 1],
+        [[0, 0], [2, 2]],
+        [[[0, 0], [0, 1]],
+         [[0, 1], [2, 2]]]
+    ]
+    expected_cdfs = [
+        [[0.75, 1.],
+         [0.25, 0.25]],
+        [[0.75, 0.25],
+         [1., 1.]],
+        [[[0.75, 0.25], [0.25, 0.25]],
+         [[0.75, 1.], [1., 1.]]]
+    ]
+
+    for event, expected_cdf in zip(events, expected_cdfs):
+      input_ = tf.convert_to_tensor(sample, dtype=np.float32)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=input_.shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph, event_ndims=1)
+      self.assertAllClose(self.evaluate(dist.cdf(event)),
+                          expected_cdf)
+
+  def testPmfNoBatch(self):
+    sample = [[0, 0], [0, 1], [0, 1], [1, 2]]
+    events = [
+        [0, 1],
+        [[0, 1], [1, 2]],
+        [[[0, 0], [0, 1]],
+         [[1, 1], [1, 2]]]
+    ]
+    expected_pmfs = [
+        0.5,
+        [0.5, 0.25],
+        [[0.25, 0.5], [0, 0.25]]
+    ]
+
+    for event, expected_pmf in zip(events, expected_pmfs):
+      input_ = tf.convert_to_tensor(sample, dtype=np.float32)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=input_.shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph, event_ndims=1)
+      self.assertAllClose(self.evaluate(dist.prob(event)),
+                          expected_pmf)
+
+  def testPmfWithBatch(self):
+    sample = [[[0, 0], [0, 1], [0, 1], [1, 1]],
+              [[0, 1], [2, 2], [2, 2], [2, 2]]]
+    events = [
+        [0, 1],
+        [[0, 0], [2, 2]],
+        [[[0, 0], [0, 1]],
+         [[0, 1], [2, 2]]]
+    ]
```

Comment:
can you add a [2,1] batch shaped event and assert a [2,2] pmf out?

Reply:
Sure I added it.